src/Provider/Provider.Web/Orchestrators/CloneVacancyOrchestrator.cs

```diff
@@ -0,0 +1,159 @@
+using System.Threading.Tasks;
+using Esfa.Recruit.Provider.Web;
+using Esfa.Recruit.Provider.Web.RouteModel;
+using Esfa.Recruit.Provider.Web.ViewModels.CloneVacancy;
+using Esfa.Recruit.Shared.Web.ViewModels;
+using Esfa.Recruit.Vacancies.Client.Domain.Exceptions;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Client;
+using Esfa.Recruit.Vacancies.Client.Domain.Extensions;
+using System;
+using Esfa.Recruit.Vacancies.Client.Domain.Entities;
+using Esfa.Recruit.Shared.Web.Orchestrators;
+using Esfa.Recruit.Vacancies.Client.Application.Providers;
+using Esfa.Recruit.Shared.Web.Extensions;
+using Microsoft.Extensions.Logging;
+using Esfa.Recruit.Vacancies.Client.Application.Validation;
+using ErrorMessages = Esfa.Recruit.Shared.Web.ViewModels.ErrorMessages;
+
+namespace Esfa.Recruit.Provider.Web.Orchestrators
+{
+    public class CloneVacancyOrchestrator : EntityValidatingOrchestrator<Vacancy, CloneVacancyWithNewDatesEditModel>
+    {
+        private const VacancyRuleSet ValidationRules = VacancyRuleSet.ClosingDate | VacancyRuleSet.StartDate | VacancyRuleSet.StartDateEndDate ;
+        const string ChangeBothDatesTitle = "Change the closing date and start date";
+        const string ChangeEitherDatesTitle = "Change the closing date or start date";
+        private readonly IRecruitVacancyClient _vacancyClient;
+        private readonly ITimeProvider _timeProvider;
+
+        public CloneVacancyOrchestrator(IRecruitVacancyClient vacancyClient,
+            ITimeProvider timeProvider, ILogger<CloneVacancyOrchestrator> logger) : base(logger)
+        {
+            _vacancyClient = vacancyClient;
+            _timeProvider = timeProvider;
+        }
+
+        public async Task<CloneVacancyDatesQuestionViewModel> GetCloneVacancyDatesQuestionViewModelAsync(VacancyRouteModel vrm)
+        {
+            var vacancy = await GetCloneableAuthorisedVacancyAsync(vrm);
+
+            if (IsNewDatesRequired(vacancy))
+                throw new InvalidStateException(string.Format(ErrorMessages.CannotCloneVacancyWithSameDates, vacancy.Title));
+
+            var vm = new CloneVacancyDatesQuestionViewModel {
```

Comment:
Is something going on? why is everyone moving to the 🤮Java curly brackets starts on same line thing when multiple things are being set.

Reply:
😬

Reply:
fixed